gitops/argocd/image/register.sh

```diff
@@ -0,0 +1,134 @@
+set -o pipefail
+
+usage() {
+
+    printf "Usage: ARGO_URL="https://argoserver.com" ARGO_USER="user" ARGO_PWD="xxxxxxxxx" DATA_DIR="/workspace" ./register.sh\n\n"
+
+    # Parameters
+    printf "The following parameters need to be passed to the script:\n"
+    printf "ARGO_URL: the address of the Argo CD server the clusters need to be registered to\n"
+    printf "ARGO_USER: the user for the authentication\n"
+    printf "ARGO_PWD: the password for the authentication\n"
+    printf "DATA_DIR: the location of the cluster files\n"
+    printf "INSECURE (optional): whether insecured connection to Argo CD should be allowed. Default value: false\n\n"
+}
+
+prechecks () {
+    if ! command -v argocd &> /dev/null; then
+        printf "Argocd CLI could not be found\n"
+    	exit 1
+    fi
+    ARGO_URL=${ARGO_URL:-}
+    if [[ -z "${ARGO_URL}" ]]; then
+        printf "ARGO_URL not set\n\n"
+        usage
+        exit 1	
+    fi
+    ARGO_USER=${ARGO_USER:-}
+    if [[ -z "${ARGO_USER}" ]]; then
+        printf "ARGO_USER not set\n\n"
+        usage
+	exit 1
+    fi
+    ARGO_PWD=${ARGO_PWD:-}
+    if [[ -z "${ARGO_PWD}" ]]; then
+	printf "ARGO_PWD not set\n\n"
+        usage
+	exit 1
+    fi
+    DATA_DIR=${DATA_DIR:-}
+    if [[ -z "${DATA_DIR}" ]]; then
+        printf "DATA_DIR not set\n\n"
+        usage
+	exit 1
+    fi
+    INSECURE=${INSECURE:-}
+    if [[ $(tr '[:upper:]' '[:lower:]' <<< "$INSECURE") == "true" ]]; then
+	printf "insecured connection to Argo CD allowed!\n"
+        insecure="--insecure"
+    else
+        insecure=""
+    fi
+}
+
+# populate clusters with the cluster names taken from the kubconfig
+# populate kubconfigs with the associated kubeconfig for each cluster name
+get_clusters() {
+    clusters=()
+    kubeconfigs=()
+    # TODO: Romain has /credentials/kubeconfig/plnsvc
+    # to clarify: is it a directory plnsvc for all the workload clusters or is plnsvc already specific to 1 workload cluster?
+    # If it is the later we need to differentiate between workload clusters (a subdirectory?), kcp and Argo CD credentials
+    # something like:
+    #   credentials/kubeconfig/workload/plnsvc1
+    #   credentials/kubeconfig/workload/plnsvc2
+    #   credentials/kubeconfig/workload/plnsvc3
+    # If it the former, why plnsvc as name? How does it identify the role they are playing within the service?
+    # Looking at /environment it seems to be the former.
+    files=($(ls $DATA_DIR/gitops/credentials/kubeconfig/plnsvc))
+    for kubeconfig in "${files[@]}"; do
+        clusters_sub=($(KUBECONFIG=${DATA_DIR}/gitops/credentials/kubeconfig/plnsvc/${kubeconfig} kubectl config get-clusters | cut -d ':' -f 1 ))
+        clusters_sub=( ${clusters_sub[@]:1} )
+        clusters+=( ${clusters_sub[@]} )
+        for i in "${!clusters_sub[@]}"; do
+            kubeconfigs+=( ${kubeconfig} )
+        done
+    done
+}
+
+parent_path=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
+
+prechecks
+
+printf "Retrieving clusters\n"
```

Comment:
Put everything into functions so the script can be sourced without being executed.

Reply:
The intent is to have the script run as part of a Tekton Pipeline, not really for it to get sourced but I can look at putting the code into functions tomorrow